subsys/net/ip/net_shell.c

```diff
@@ -182,10 +189,37 @@ static void iface_cb(struct net_if *iface, void *user_data)
 		return;
 	}
 
-	printk("Link addr : %s\n",
-	       net_sprint_ll_addr(net_if_get_link_addr(iface)->addr,
-				  net_if_get_link_addr(iface)->len));
+#if defined(CONFIG_NET_VLAN)
```

Comment:
iface_cb is getting quite bloated. Maybe it will be worth at some point to split it by centralizing each part into a dedicated inline function (one for vlan/ethernet, one for ipv6, etc...). Not it this PR though